code/parachain/runtime/picasso/src/ibc.rs

```diff
@@ -33,21 +33,47 @@ impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
 	type Error = DispatchError;
 
 	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
-		ForeignIbcIcs20Assets::<AssetsRegistry>::from_denom_to_asset_id(denom)
+		log::debug!("converting denom {} to currency id", denom);
+		let denom_bytes = denom.as_bytes().to_vec();
+		if let Some(id) = IbcDenoms::<Runtime>::get(&denom_bytes) {
+			log::debug!("converted {:} to currency id", &id);
+			return Ok(id)
+		}
+
+		// +1 nonce
+		let asset_id =
+			<currency_factory::Pallet<Runtime> as CurrencyFactoryT>::create(RangeId::IBC_ASSETS)?;
+
+		let asset_id = <assets_transactor_router::Pallet<Runtime> as composable_traits::assets::CreateAsset>
+		::create_local_asset([0;8], asset_id.0 as u64,
+			AssetInfo {
+				name: None,
+				symbol: None,
+				decimals: None,
+				// we do not have sufficient assets, nor setting zero is good
```

Comment:
This may need a more rigorous explanation, why is setting zero not good? Are these `10_000` simply minted into existence here?

Reply:
I replaced this trait implementation on the implementation from `ibc.rs`.
The comments aren't relevant already.